src/overlays/actors/ovl_Arrow_Light/z_arrow_light.c

```diff
@@ -34,34 +36,201 @@ const ActorInit Arrow_Light_InitVars = {
+    this->radius = 0;
+    this->height = 1.0f;
+    ArrowLight_SetupAction(this, ArrowLight_Charge);
+    Actor_SetScale(&this->actor, 0.01f);
+    this->alpha = 130;
+    this->timer = 0;
+    this->blueingEffectMagnitude = 0.0f;
+}
+
+void ArrowLight_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    func_80115D5C(&globalCtx->state);
+    (void)"消滅";
+}
+
+void ArrowLight_Charge(ArrowLight* this, GlobalContext* globalCtx) {
+    EnArrow* arrow = (EnArrow*)this->actor.parent;
+
+    if ((arrow == NULL) || (arrow->actor.update == NULL)) {
+        Actor_MarkForDeath(&this->actor);
+        return;
+    }
+
+    if (this->radius < 10) {
+        this->radius++;
+    }
+
+    this->actor.world.pos = arrow->actor.world.pos;
+    this->actor.shape.rot = arrow->actor.shape.rot;
+
+    func_800B9010(&this->actor, NA_SE_PL_ARROW_CHARGE_LIGHT - SFX_FLAG);
+    if (arrow->actor.parent == NULL) {
+        this->firedPos = this->actor.world.pos;
+        this->radius = 10;
+        ArrowLight_SetupAction(this, ArrowLight_Fly);
+        this->alpha = 255;
+    }
+}
+
+void ArrowLight_Lerp(Vec3f* firedPos, Vec3f* pos, f32 scale) {
+    VEC3F_LERPIMPDST(firedPos, firedPos, pos, scale);
+}
+
+void ArrowLight_Hit(ArrowLight* this, GlobalContext* globalCtx) {
+    f32 scale;
+    u16 timer;
+
+    if (this->actor.projectedW < 50.0f) {
+        scale = 10.0f;
+    } else if (this->actor.projectedW > 950.0f) {
+        scale = 310.0f;
+    } else {
+        scale = this->actor.projectedW;
+        scale = ((scale - 50.0f) * (1.0f / 3.0f)) + 10.0f;
+    }
+
+    timer = this->timer;
+    if (timer != 0) {
+        this->timer--;
+        if (this->timer >= 8) {
+            f32 offset = (this->timer - 8) * (1.0f / 24.0f);
+
+            offset = SQ(offset);
+            this->radius = (((1.0f - offset) * scale) + 10.0f);
+            this->height = this->height + ((2.0f - this->height) * 0.1f);
+            if (this->timer < 16) {
+                this->alpha = (this->timer * 35) - 280;
+            }
+        }
+    }
+    if (this->timer >= 9) {
+        if (this->blueingEffectMagnitude < 1.0f) {
+            this->blueingEffectMagnitude += 0.25f;
+        }
+    } else {
+        if (this->blueingEffectMagnitude > 0.0f) {
+            this->blueingEffectMagnitude -= 0.125f;
+        }
+    }
+    if (this->timer < 8) {
+        this->alpha = 0;
+    }
+    if (this->timer == 0) {
+        this->timer = 255;
+        Actor_MarkForDeath(&this->actor);
+        return;
+    }
+}
+
+void ArrowLight_Fly(ArrowLight* this, GlobalContext* globalCtx) {
+    EnArrow* arrow = (EnArrow*)this->actor.parent;
+    s32 pad;
+    s32 pad2;
+
+    if ((arrow == NULL) || (arrow->actor.update == NULL)) {
+        Actor_MarkForDeath(&this->actor);
+        return;
+    }
+
+    this->actor.world.pos = arrow->actor.world.pos;
+    this->actor.shape.rot = arrow->actor.shape.rot;
+
+    this->height = Math_Vec3f_DistXYZ(&this->firedPos, &this->actor.world.pos) * 0.041666668f;
```

Comment:
```suggestion
    this->height = Math_Vec3f_DistXYZ(&this->firedPos, &this->actor.world.pos) * (10.0f * (1.0f / 240.0f));
```
While it's clear that `(1.0f / 24.0f)` likely won't match, we've found recently that when it's slightly off of what you'd expect, it can sometimes means there's a factor of 10 or 100 in there that's causing it to be off. So this is just an idea to try. Or also `(100.0f * (1.0f / 2400.0f))`.

Reply:
`1.0f / 24.0f` did match